src/components/entity/ha-entity-attribute-picker.ts

```diff
@@ -150,6 +156,8 @@ class HaEntityAttributePicker extends LitElement {
   }
 
   private _setValue(value: string) {
+    // Ensure that we send the correct internal snake_case name to the callbacks / events
+    value = formatAttributeNameIn(value);
```

Comment:
We should not do this, we should keep the true value and not calculate back and forth.

Reply:
I am not sure I understand your point. To not break / disrupt the calling card / editor / ... that uses the picker we have to provide back the internal snake_case attribute name.

Is your point coming from a performance perspective in the sense that we should ensure we only convert the snake_case values to the user friendly values once when the picker is first shown (plus if the items change) and then keep them in a class member. And then look up the matching value in a dict so we do not have to convert it back later, but just need a simple look-up instead?

Reply:
I have multiple problems with it, and yes one is performance. The other would be that we might get in a position that we can't simply reverse the process to get the same value back.

Can't we just add the attribute name as a property we can access on the selected event?

Reply:
We can use `itemLabelPath` and `itemValuePath`? <https://www.webcomponents.org/element/vaadin/vaadin-combo-box/elements/vaadin-combo-box-light#property-itemLabelPath>

Reply:
I am not sure if those vaadin properties will work with the input field. Getting nicer values rendered in the dropdown can also easily be done via the vaadin renderer function (I used that at the beginning). But then we still ended up with the internal name in the input field after the user had selected a value from the dropdown menu.